invoice.py

```diff
@@ -51,27 +51,48 @@ def __register__(cls, module_name):
         Identifier = pool.get('party.identifier')
         identifier = Identifier.__table__()
 
+        exist_tax_identifier = table.column_exist('tax_identifier')
         exist_party = table.column_exist('party')
+
+        if exist_tax_identifier:
```

Comment:
You're always droping tax_identifier field but you cannot do that because you did not rename the column.

We could probably check the type of "tax_identifier" field but maybe it is easier to rename the field. That is: 

- remove existing "tax_identifier = fields.Char(..)"
- add the new field as "party_tax_identifier = fields.Many2One('party.identifier', 'Party Tax Identifier')

this would make the name consistent with account.invoice and make the implementation of register much easier.
Maybe we should check the *type* of tax_identifier field, for example.

Reply:
Done.
Use the party_tax_identifier as in account.invoice.